### PR TITLE
Factoring out the quantized mappings

### PR DESCRIPTION
**Pull request by z-a-f**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#23351 Factoring out the quantized mappings**

Differential Revision: [D16470907](https://our.internmc.facebook.com/intern/diff/D16470907)

### CODE REVIEW

File: torch/quantization/quantize.py [hunk 1]
```diff
@@ -4,6 +4,9 @@
 import torch.nn.qat as qat
```
Comment (jerryzh168) [inline torch/quantization/quantize.py]: then we can remove import of nnq and qat?

File: torch/quantization/_mappings.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import torch
+
+from .quantize import QuantStub
```
Comment (jerryzh168) [inline torch/quantization/_mappings.py]: do we have circular dependency here?

File: torch/quantization/_mappings.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+from .quantize import DeQuantStub
+
+# Map for swapping float module to quantized ones.
+_DEFAULT_MODULE_MAPPING = {
+    torch.nn.Linear: torch.nn.quantized.Linear,
```
Comment (jerryzh168) [inline torch/quantization/_mappings.py]: we can import nn, nnq and qat here to simply namespaces.

**Comment by zafartahirov** [comment]
Closing in favor of #27164

**Pull request closed without merging**